src/FileConventions/Library.fs

```diff
@@ -104,3 +104,100 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let IsFooterReference(line: string) : bool =
+    line.[0] = '[' && line.IndexOf "] " > 0
+
+let IsFixesOrClosesSentence(line: string) : bool =
+    line.IndexOf "Fixes " = 0 || line.IndexOf "Closes " = 0
+
+let IsCoAuthoredByTag(line: string) : bool =
+    line.IndexOf "Co-authored-by: " = 0
+
+let IsFooterNote(line: string) : bool =
+    IsFooterReference line
+    || IsCoAuthoredByTag line
+    || IsFixesOrClosesSentence line
+
+type Word =
+    | CodeBlock
+    | FooterNote
+    | PlainText
+
+type Text(textStr: string, textType: Word) =
+    member this.Type = textType
+    member this.Text = textStr
+
+    member this.Length() =
+        String.length this.Text
+
+    override self.ToString() =
+        sprintf "Type: %A, Text: %s" self.Type self.Text
+
+let SplitIntoWords(text: string) =
+    let codeBlockRegex = "\s*(```[\s\S]*```)\s*"
+
+    let words =
+        Regex.Split(text, codeBlockRegex)
+        |> Seq.filter(fun item -> not(String.IsNullOrEmpty item))
+        |> Seq.map(fun item ->
+            if Regex.IsMatch(item, codeBlockRegex) then
+                Text(item, CodeBlock)
+            else
+                Text(item, PlainText)
+        )
+        |> Seq.map(fun paragraph ->
+            if paragraph.Type = CodeBlock then
+                Seq.singleton paragraph
+            else
+                let lines = paragraph.Text.Split(Environment.NewLine)
+
+                lines
+                |> Seq.map(fun line ->
+                    if IsFooterNote line then
+                        Seq.singleton(Text(line, FooterNote))
+                    else
+                        line.Split(" ")
+                        |> Seq.map(fun word -> Text(word, PlainText))
+                )
+                |> Seq.concat
+        )
+        |> Seq.concat
+
+    words |> Seq.toList
+
+let WrapParagraph (paragraph: string) (maxCharsPerLine: int) : string =
```

Comment:
@realmarv this func should be private

Reply:
How should I test a private function?

Reply:
you shouldn't test WrapParagraph, you should just test WrapText

Reply:
Done